### PR TITLE
Bug 1156650 - Show display dialog when launching app from homescreen

### PR DESCRIPTION
**Pull request by luke-chang**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b5651e7277b60f964fe84c988a392ff315af5c4a

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=799d783167b2b32459139f53ef16feb57a27296c

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=77cf67bd79b923c90090a82c76afea831d64e2e3

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=96f724ecbad96e6233dc77c6af6e68339d71bb43

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b05f2ba465d666c11d0ca49331fc7d3044fcaa6f

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5a75599a5d2b89b1b475da8517afeeb6a1422da4

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3877595758cb39418a893a414bdfe86cacc0e1d9

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a6e8e7ae4e67aa89a59172df751a65d9a2bbdcc6

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=420eff0c02c9ed673cc3b2493c45989791eb5210

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f69e58082cee390cbeca3e520cdc28073260df74

File: apps/system/test/unit/multi_screen_controller_test.js [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+  function triggerMozChromeEvent(detail) {
+    window.dispatchEvent(new CustomEvent('mozChromeEvent', {
+      detail: detail
+    }));
+  }
+
+  setup(function() {
+    subject = BaseModule.instantiate('MultiScreenController');
+    subject.start();
+    subject._enabled = true;
```
Comment (alivedise) [inline apps/system/test/unit/multi_screen_controller_test.js]: you should test trigger settings change will cause this value to change

File: apps/system/test/unit/app_window_factory_test.js [hunk 2]
```diff
@@ -10,6 +10,7 @@
 
 'use strict';
 
+requireApp('system/shared/test/unit/mocks/mock_service.js');
```
Comment (alivedise) [inline apps/system/test/unit/app_window_factory_test.js]: Need to test chooseDisplay is called here if multiscreen.enabled is true

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4f0aeffe470ffa30c6d32b7abf671fecec5eaa4c

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=749f0971c60f616b1ad73a73e1004a12d04abb97

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=501c1993f7192ab4c5d55098885d8c7f2df40008

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=992aba341738e4ebb9968ffa06606c02f4f841ec

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=7a7e83e31b9c72fec88d5062dc2609af622c88f0

**Pull request merged**